### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/installation.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/installation.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/installation.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+قم باستنساخ المستودع وقم بتثبيت 🤗 Transformers باستخدام الأوامر التالية:
+
+```bash
+git clone https://github.com/huggingface/transformers.git
+cd transformers
+pip install -e .
+```
+
+سترتبط هذه الأوامر مجلد المستودع الذي قمت باستنساخه ومسارات مكتبة Python. الآن، سيبحث Python داخل المجلد الذي قمت باستنساخه بالإضافة إلى مسارات المكتبة العادية. على سبيل المثال، إذا تم تثبيت حزم Python الخاصة بك عادةً في `~/anaconda3/envs/main/lib/python3.7/site-packages/`, فسيقوم Python أيضًا بالبحث في المجلد الذي قمت باستنساخه: `~/transformers/`.
```
Comment (abodacs) [inline docs/source/ar/installation.md:118]:
```suggestion
 ستقوم هذه الأوامر بربط المجلد الذي قمت باستنساخ المستودع فيه بمسارات مكتبة Python. بمعنى آخر، سيبحث Python داخل المجلد الذي قمت باستنساخه بالإضافة إلى المسارات المعتادة للمكتبات. على سبيل المثال، إذا تم تثبيت حزم Python الخاصة بك عادةً في `~/anaconda3/envs/main/lib/python3.7/site-packages/`, فسيقوم Python أيضًا بالبحث في المجلد الذي قمت باستنساخه: `~/transformers/`.
```

File: docs/source/ar/installation.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+قم بالتثبيت من قناة conda `conda-forge`:
+
+```bash
+conda install conda-forge::transformers
+```
+
+## إعداد ذاكرة التخزين المؤقت
+
+يتم تنزيل النماذج المدربة مسبقًا وتخزينها مؤقتًا محليًا في: `~/.cache/huggingface/hub`. هذا هو دليل ذاكرة التخزين المؤقت الافتراضية المقدمة بواسطة متغير البيئة `TRANSFORMERS_CACHE`. على Windows، يكون دليل ذاكرة التخزين المؤقت الافتراضي هو `C:\Users\username\.cache\huggingface\hub`. يمكنك تغيير متغيرات البيئة أدناه - حسب الأولوية - لتحديد دليل ذاكرة التخزين المؤقت المختلفة:
```
Comment (abodacs) [inline docs/source/ar/installation.md:145]:
```suggestion
تُحمّل النماذج المُسبقة التدريب وتُخزّن مؤقتًا في: `~/.cache/huggingface/hub`. هذا هو المجلد الافتراضي الذي يُحدده متغير البيئة `TRANSFORMERS_CACHE`. على Windows، يكون دليل ذاكرة التخزين المؤقت الافتراضي هو `C:\Users\username\.cache\huggingface\hub`. يمكنك تغيير متغيرات البيئة shell الموضحة أدناه - حسب الأولوية - لتحديد دليل ذاكرة تخزين مؤقت مختلف:
```

File: docs/source/ar/installation.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+2. متغير البيئة: `HF_HOME`.
+3. متغير البيئة: `XDG_CACHE_HOME` + `/huggingface`.
+
+<Tip>
+
+سيستخدم 🤗 Transformers متغيرات البيئة `PYTORCH_TRANSFORMERS_CACHE` أو `PYTORCH_PRETRAINED_BERT_CACHE` إذا كنت قادمًا من إصدار سابق من هذه المكتبة وقمت بتعيين متغيرات البيئة هذه، ما لم تحدد متغير البيئة `TRANSFORMERS_CACHE`.
+
+</Tip>
+
+## الوضع غير المتصل
```
Comment (abodacs) [inline docs/source/ar/installation.md:157]:
```suggestion
## الوضع دون اتصال بالإنترنت
```

File: docs/source/ar/installation.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+أضف [🤗 Datasets](https://huggingface.co/docs/datasets/) إلى سير عمل التدريب غير المتصل باستخدام متغير البيئة `HF_DATASETS_OFFLINE=1`.
+
+</Tip>
+
+```bash
+HF_DATASETS_OFFLINE=1 HF_HUB_OFFLINE=1 \
+python examples/pytorch/translation/run_translation.py --model_name_or_path google-t5/t5-small --dataset_name wmt16 --dataset_config ro-en ...
+```
+
+يجب أن يعمل هذا البرنامج النصي دون تعليق أو انتظار انتهاء المهلة لأنه لن يحاول تنزيل النموذج من Hub.
```
Comment (abodacs) [inline docs/source/ar/installation.md:172]:
```suggestion
يجب أن يعمل هذا البرنامج النصي دون توقف أو انتظار انتهاء المهلة الزمنية لأنه لن يحاول تنزيل النموذج من Hub.
```

File: docs/source/ar/installation.md [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+يمكنك أيضًا تجاوز تحميل نموذج من Hub من كل مكالمة [`~PreTrainedModel.from_pretrained`] باستخدام معلمة [`local_files_only`]. عندما يتم تعيينه على `True`، يتم تحميل الملفات المحلية فقط:
```
Comment (abodacs) [inline docs/source/ar/installation.md:174]:
```suggestion
يمكنك أيضًا تجاوز تحميل نموذج من Hub من كل استدعاء [`~PreTrainedModel.from_pretrained`] باستخدام معلمة [`local_files_only`]. عندما يتم تعيينها على `True`، يتم تحميل الملفات المحلية فقط:
```

File: docs/source/ar/installation.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+```py
+from transformers import T5Model
+
+model = T5Model.from_pretrained("./path/to/local/directory", local_files_only=True)
+```
+
+### الحصول على النماذج ومعالجات التوكينز لاستخدامها دون اتصال
```
Comment (abodacs) [inline docs/source/ar/installation.md:182]:
```suggestion
### جلب النماذج والمُجزّئات لاستخدامها دون اتصال بالإنترنت
```

File: docs/source/ar/installation.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+    2. استخدم وظيفة [`hf_hub_download`](https://huggingface.co/docs/hub/adding-a-library#download-files-from-the-hub) لتنزيل ملف إلى مسار محدد. على سبيل المثال، يقوم الأمر التالي بتنزيل ملف `config.json` من نموذج [T0](https://huggingface.co/bigscience/T0_3B) إلى المسار المطلوب:
+
+    ```py
+    >>> from huggingface_hub import hf_hub_download
+
+    >>> hf_hub_download(repo_id="bigscience/T0_3B", filename="config.json", cache_dir="./your/path/bigscience_t0")
+    ```
+
+بمجرد تنزيل ملفك وتخزينه مؤقتًا محليًا، حدد مساره المحلي لتحميله واستخدامه:
```
Comment (abodacs) [inline docs/source/ar/installation.md:234]:
```suggestion
بمجرد تنزيل ملفك وتخزينه مؤقتًا محليًا، حدد مساره المحلي الخاص به لتحميله واستخدامه:
```

**Review by abodacs — CHANGES_REQUESTED** [review]

**Review by abodacs — CHANGES_REQUESTED** [review]
@AhmedAlmaghz 
شكرا على مجهودك

تمت المراجعة

File: docs/source/ar/installation.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
+## التثبيت
```
Comment (stevhliu) [inline docs/source/ar/installation.md:1]: This should be a top-level header, `# Installation`

**Review by stevhliu — COMMENTED** [review]
Thanks! Once the `toctree` is merged in your other PR, you'll need to add this doc to the `toctree`.

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs  @stevhliu

**Comment by AhmedAlmaghz** [comment]
@abodacs 
Next :
- [ ]  Run inference with pipelines #33065
- [ ]  Write portable code with AutoClass #33022
- [ ]  Preprocess data #33068
- [ ]  Fine-tune a pretrained model #33081
- [ ]  Train with a script #33070
- [ ]  Set up distributed training with 🤗 Accelerate #33016
- [ ]  Load and train adapters with 🤗 PEFT #33051
- [ ]  Share your model #33046
- [ ]  Agents #33020
- [ ]  Generation with LLMs #33044
- [ ] Chatting with Transformers #33029

**Pull request closed without merging**